packages/headless/src/features/search/search-slice.ts

```diff
@@ -17,7 +23,7 @@ function handleFulfilledSearch(
   action: ReturnType<SearchAction['fulfilled']>
 ) {
   state.error = null;
-  state.response = action.payload.response;
+  state.response = shimResponsePayload(action.payload.response);
```

Comment:
If the shim occurs in inside the action instead of inside the slice, then the full payload would be available to all cases handling executeSearch.fulfilled.

Reply:
Indeed, I just realized it's probably not the correct place to do this.

Another approach would be to do this directly at the response level (ie: inside search-api-client or something like that).

I think I prefer to do it there actually (inside search api client)... It would also be available for any external code using postProcess middleware.

Reply:
Indeed, the search api client would be even better than the action,